File: code/C++/Chapter8/8-2/inverses.cpp
#include <iostream>

using namespace std;

int count_inverses(int *arr, int n) {
    int count = 0;
    for (int i = 0; i < n; i++) {
        for (int j = i + 1; j < n; j++) {
            if (arr[i] > arr[j]) {
                count++;
            }
        }
    }
    return count;
}

int main() {
    int arr[] = {5, 8, 6, 3, 9, 2, 1, 7};
    int n = sizeof(arr) / sizeof(arr[0]);

    cout << count_inverses(arr, n) << endl;

    return 0;
}

File: code/Java/Chapter8/8-2/Inverses.java
public class Inverses {
    public static void main(String[] args) {
        int[] arr = {5, 8, 6, 3, 9, 2, 1, 7};
        System.out.println(countInverses(arr));
    }

    public static int countInverses(int[] arr) {
        int count = 0;
        for (int i = 0; i < arr.length; i++) {
            for (int j = i + 1; j < arr.length; j++) {
                if (arr[i] > arr[j]) {
                    count++;
                }
            }
        }

        return count;
    }
}
